parla/tasks.py

```diff
@@ -341,7 +342,10 @@ def spawn(taskid: Optional[TaskID] = None, dependencies = (), *,
           placement: Union[Collection[PlacementSource], Any, None] = None,
           ndevices: int = 1,
           tags: Collection[Any] = (),
-          data: Collection[Any] = None
+          data: Collection[Any] = None,
+          input: Collection[Any] = (),
+          output: Collection[Any] = (),
+          inout: Collection[Any] = ()
```

Comment:
just curious as a Python novice. What is difference between () and None? If they are the same, how about just matching to one?

Reply:
() make it a empty tuple by default, which is iterable. 
There is a chain(self._input, self._output, self._inout) in dataflow.py:25 which cat them and iterate over them. An error will be raised it one of them is none.